src/Driver.cpp

```
@@ -55,64 +66,149 @@ int Driver::parsePacket(uint8_t const* buffer, size_t size)
+    
 }
 
-void Driver::parseReadSingleIOReply(uint8_t const* buffer, size_t size)
+void Driver::parseReadSingleIOReply(uint8_t const* buffer)
 {
     msgs::ReadSingleIoReplyMsg const& msg = *reinterpret_cast<msgs::ReadSingleIoReplyMsg const*>(buffer);
-}
\ No newline at end of file
+}
+
+void Driver::sendJointTrajPTFullCmd(int robot_id, int sequence, base::Time timestamp,
+                                    std::vector<base::JointState> joint_states)
+{
+    msgs::JointTrajPTFullMsg joint_traj_cmd;
+    joint_traj_cmd.prefix.length = 46; 
+    joint_traj_cmd.prefix.msg_type = MotomanMsgTypes::MOTOMAN_JOINT_TRAJ_PT_FULL;
+    joint_traj_cmd.robot_id = int32_t(robot_id);
+    joint_traj_cmd.sequence = int32_t(sequence);
+    joint_traj_cmd.time = timestamp.toSeconds();
+    for(int i=0; i<joint_states.size();i++)
+    {
+        joint_traj_cmd.positions[i] =  joint_states[i].position;
+        joint_traj_cmd.velocities[i] = joint_states[i].speed;
+        joint_traj_cmd.accelerations[i] = joint_states[i].acceleration;
+    }
+    
+    uint8_t const* buffer = reinterpret_cast<uint8_t const*>(&joint_traj_cmd);
+    writePacket(buffer, joint_traj_cmd.prefix.length + 4);
+    readJointFeedback(base::Time::fromSeconds(0.1));
+}
+
+bool Driver::waitForReply(base::Time const& timeout, int32_t msg_type)
+{
+    base::Timeout deadline = base::Timeout(timeout);
+    
+    while(deadline.timeLeft().toMicroseconds() > .0)
+    {
+        int packet_size = readPacket(&buffer[0], 10000, deadline.timeLeft());
+        if(packet_size != returnMsgSize(msg_type))
+            continue;
+        int32_t const* buffer_as_int32 = reinterpret_cast<int32_t const*>(&buffer[0]);
+        if(buffer_as_int32[1]!= msg_type)
+            continue;
+        else
+            return true;
+    }
+    return false;
+}
+
+void Driver::readJointFeedback(base::Time const& timeout)
+{
+    if(waitForReply(timeout,MotomanMsgTypes::MOTOMAN_JOINT_FEEDBACK))
+        parseJointFeedback(&buffer[0]);
```

Actually, the best would be to return the parsed structure all the way up. Since you will now throw in `waitForReply` instead of ignoring a missed reply, you can do that.